plugin/compiler-plugin/src/main/kotlin/it/unibo/collektive/utils/branch/BranchAlignment.kt

```diff
@@ -135,53 +78,42 @@ private fun IrBuilderWithScope.buildLambda(
+        is IrBlock -> expression.statements
+        else -> mutableListOf()
     }
-}
-
-private fun IrBuilderWithScope.buildLambda(
-    pluginContext: IrPluginContext,
-    expression: IrBlock,
-): IrSimpleFunction = pluginContext.irFactory.buildFun {
-    name = Name.special("<anonymous>")
-    this.origin = IrDeclarationOrigin.LOCAL_FUNCTION_FOR_LAMBDA
-    this.visibility = DescriptorVisibilities.LOCAL
-}.apply {
-    this.patchDeclarationParents(this@buildLambda.parent)
-    val lastExpression = expression.statements.removeLast() as IrExpression
-    this.returnType = getReturnType(lastExpression)
-    this.body = context.irBuiltIns.createIrBuilder(symbol).irBlockBody {
-        for (bodyStatement in expression.statements) {
+    val lastExpression = (statements.removeLastOrNull() ?: expression) as IrExpression
+    returnType = getReturnType(lastExpression)
+    body = context.irBuiltIns.createIrBuilder(symbol).irBlockBody {
+        for (bodyStatement in statements) {
             +bodyStatement
         }
         +irReturn(lastExpression)
     }
 }
 
-private fun conditionName(condition: IrExpression): String {
-    return when (condition) {
-        is IrGetValue -> condition.symbol.owner.name.asString()
-        is IrConst<*> -> "constant"
-        is IrTypeOperatorCall -> condition.operator.name + " " + condition.typeOperand.classFqName // 'is' in the 'when'
-        is IrCall -> condition.symbol.owner.name.asString() +
-            (condition.dispatchReceiver?.let { " " + conditionName(it) } ?: "")
-        is IrWhen -> conditionName(condition.branches[0].condition) + when {
-            condition.origin == IrStatementOrigin.ANDAND -> " & " + conditionName(condition.branches[0].result)
-            else -> " | " + conditionName(condition.branches[1].result)
-        }
-
-        else -> error(
-            "The current if condition type ${condition::class} has not been handled for the alignment yet. " +
-                "Update the compiler plugin.",
-        )
-    }
-}
+// private fun conditionName(condition: IrExpression): String {
+//    return when (condition) {
+//        is IrGetValue -> condition.symbol.owner.name.asString()
+//        is IrConst<*> -> "constant"
+//        is IrTypeOperatorCall -> condition.operator.name + " " + condition.typeOperand.classFqName
+//        is IrCall -> condition.symbol.owner.name.asString() +
+//            (condition.dispatchReceiver?.let { " " + conditionName(it) } ?: "")
+//        is IrWhen -> conditionName(condition.branches[0].condition) + when {
+//            condition.origin == IrStatementOrigin.ANDAND -> " & " + conditionName(condition.branches[0].result)
+//            else -> " | " + conditionName(condition.branches[1].result)
+//        }
+//
+//        else -> error(
+//            "The current if condition type ${condition::class} has not been handled for the alignment yet. " +
+//                "Update the compiler plugin.",
+//        )
+//    }
+// }
```

Comment:
Do we need these lines to remain commented? Can't we drop them?

Reply:
Yes, we can!